hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -533,8 +533,18 @@ public void createBucket(
     verifyCountsQuota(bucketArgs.getQuotaInNamespace());
     verifySpaceQuota(bucketArgs.getQuotaInBytes());
 
-    String owner = bucketArgs.getOwner() == null ?
-            ugi.getShortUserName() : bucketArgs.getOwner();
+    String owner;
+    // If S3 auth exists, set owner name to the short user name derived from the
+    //  accessId. Similar to RpcClient#getDEK
+    if (getThreadLocalS3Auth() != null) {
+      UserGroupInformation s3gUGI = UserGroupInformation.createRemoteUser(
+          getThreadLocalS3Auth().getAccessID());
```

Comment:
Will there be a follow up to fix this in the s3 Multi Tenancy branch? (Given the accessId is NOT the actual user name)

Reply:
Yes. In multi-tenancy branch we will just need to replace `getThreadLocalS3Auth().getAccessID()` with `getThreadLocalS3Auth().getUserPrincipal()` instead.